src/components/Match/MatchStory.jsx

```diff
@@ -691,9 +717,19 @@ const generateStory = (match) => {
   }
 
   // Chat messages
+  let lastMessage;
   const chatMessageEvents = match.chat
     .filter(obj => obj.type === 'chat')
-    .map(obj => new ChatMessageEvent(match, obj));
+    .map((obj) => {
+      let event;
+      if (lastMessage !== undefined) {
+        event = new ChatMessageEvent(match, obj, lastMessage);
+      } else {
+        event = new ChatMessageEvent(match, obj);
+      }
+      lastMessage = event;
```

Comment:
maybe do `.map(obj, i)` and then use `i` and `i - 1` to find the last message?  That removes the dependence on state outside the function

Reply:
Yeah good call, much better scope handling.